R/get_stan_data.R

```diff
@@ -351,6 +329,16 @@ get_stan_data <- function(input_count_data,
 
   inf_to_count_delay_max <- length(inf_to_count_delay)
 
+  # if adding auxiliary site, then we need to adjust site and lab indices
+  # such that first subpopulation is the auxiliary site.
+  if (add_auxiliary_site) {
+    subpop_to_samples_map <- ww_indices$ww_sampled_sites + 1
+    lab_site_to_subpop_map <- ww_indices$lab_site_to_site_map + 1
+  } else {
+    subpop_to_samples_map <- ww_indices$ww_sampled_sites
+    lab_site_to_subpop_map <- ww_indices$lab_site_to_site_map
+  }
```

Comment:
Are you saying an additional `site_to_subpop_map` as well as the `labs_site_to_subpop_map`. I would prefer not to remove the `lab_site_to_subpop_map` because this is the unit that maps latent infections to observations, whereas `site_to_subpop` only gets you somewhere intermediate.... (you'd have to additionally index site_to_lab_site)

So the mapping would look something like:
`exp_obs_conc[lab_site[i], t] = conc[site_to_subpop[site_to_lab[i]], t]` vs what we have now which is:
`exp_obs_conc[lab_site[i], t] = conc[lab_site_to_subpop[i]]`

Reply:
Ok see my update:
```r
 # This creates a dataframe that maps subpopulations to sample indices
  # by joining the site:subpop map to vector of site indices for each sample.
  # Then we just pull the subpop indices corresponding to each sample
  subpop_to_samples_map <- tibble::tibble(
    site_index = ww_indices$ww_sampled_sites
  ) |>
    dplyr::left_join(subpop_data$site_to_subpop_map, by = "site_index") |>
    pull(subpop_index)
  # This creates a dataframe that maps subpopulations to lab_site indices
  # by joining the site:subpop map to lab_site indices. Then we just pull the
  # subpop indices corresponding to the lab_sites
  lab_site_to_subpop_map <- tibble::tibble(
    site_index = ww_indices$lab_site_to_site_map
  ) |>
    dplyr::left_join(subpop_data$site_to_subpop_map, by = "site_index") |>
    pull(subpop_index)
```

What I am doing here is making a tibble in `get_subpop_data()` that looks like:
```
# A tibble: 5 × 2
  site_index subpop_index
       <int>        <int>
1         NA            1
2          1            2
3          2            3
4          3            4
5          4            5
```
Then I am using it to get the indices I want here by joining this to the wastewater data by site index and passing the vector of correspinding subpop indices.

My proposal is that since this mapping + the `site_name` is needed in post processing + a mapping of dates to times and a mapping of subpops to lab_sites, I think these 3 mappings should be added to the output of wwinference. This way, nothing in postprocessing needs to be redone if we change how we do subpopulations or if there are multiple subpopulations without site level observations etc. 

The 3 mappings would be:
```
site_to_subpop_map <- 
site_index | subpop_index | subpop_name | site_name 

lab_site_to_subpop_map <-
lab_site_index | subpop_index | subpop_name | lab_site_name

date_to_t_map <-
t | date